core/modules/util/IterableFormatter.h

```diff
@@ -0,0 +1,125 @@
+ * the GNU General Public License along with this program.  If not,
+ * see <http://www.lsstcorp.org/LegalNotices/>.
+ */
+#ifndef LSST_QSERV_UTIL_CONTAINERFORMATTER_H
+#define LSST_QSERV_UTIL_CONTAINERFORMATTER_H
+ /**
+  * @file
+  *
+  * @brief  Misc. lightweight vector manipulation.
+  *
+  * @author Fabrice Jammes, IN2P3/SLAC
+  */
+
+// System headers
+#include <vector>
+#include <iterator>
+#include <sstream>
+#include <string>
+
+// Third-party headers
+
+// LSST headers
+
+// Qserv headers
+
+namespace lsst {
+namespace qserv {
+namespace util {
+
+/**
+ *  Provide generic output operator for iterable data structures
+ *
+ *  Output format is [a, b, c, ...]
+ *  The elements stored in the iterable data structure
+ *  must provide an ouput operator.
+ *  Examples:
+ *  - works fine for std::vector<std::string>,
+ *  - doesn't work for std::multimap.
+ */
+template <typename Iterable>
+class IterableFormatter
+{
+public:
+
+    /**
+     *  Create a printable wrapper for an iterable data structure
+     *
+     *  @param x:       an iterable data structure
+     *  @param first:   start printing from x[first]
+     *  @param open:    opening bracket, NULL is undefined behaviour
+     *  @param close:   closing bracket, NULL is undefined behaviour
+     *  @param sep:     separator between elements, NULL is undefined behaviour
+     */
+    explicit IterableFormatter(const Iterable& x, int first, char const *const open, char const *const close,
+                               char const *const sep) :
+            _ref(x), _first(first), _open(open), _close(close), _sep(sep) {
+    }
+
+    /**
+     *  Output operator for IterableFormatter<Iterable>
+     *
+     *  @param out
+     *  @param iterableFormatter
+     *  @return an output stream, with no newline at the end
+     */
+    friend std::ostream& operator<<(std::ostream& os,
+                                    const IterableFormatter<Iterable>& self) {
+        os << self._open;
+        size_t last = self._ref.size() - 1;
+        for(size_t i=self._first; i<self._ref.size(); ++i) {
+            os << self._ref[i];
+            if (i != last)
+                os << self._sep << " ";
+        }
+        os << self._close;
+        return os;
+    }
+
+private:
+    Iterable const& _ref;
+    int const _first;
+    char const *const _open;
+    char const *const _close;
+    char const *const _sep;
+};
+
+/**
+ *  Create a printable wrapper for an iterable data structure
+ *
+ *  @param x:       an iterable data structure
+ *  @param first:   start printing from x[first]
+ *  @param open:    opening bracket, NULL is undefined behaviour
+ *  @param close:   closing bracket, NULL is undefined behaviour
+ *  @param sep:     separator between elements, NULL is undefined behaviour
+ *  @return:        an object wrapping x and providing an output operator
+ */
+template <typename Iterable>
+IterableFormatter<Iterable> formatable(Iterable const& x, int first = 0,
```

Comment:
Another thing - if you do not want to print whole container, then the standard (and more flexible) C++ solution is to use couple of iterators instead of using `first` index. WOuld it be better to define couple of these methods:

    template <typename Iter>
    IterableFormatter<Iter> formattable(Iter begin, Iter end, char const *const open, ...) { ... }

    template <typename Iterable>
    IterableFormatter<typename Iterable::const_iterator> formattable(Iterable x, char const *const open, ...) {
        return formattable(x.begin(), x.end(), open, ...);
    }

IterableFormatter has to be changed of course to work with iterators.